dart-if/include/dash/dart/if/dart_globmem.h

```diff
@@ -257,6 +257,89 @@ dart_ret_t dart_gptr_setflags(
   dart_gptr_t * gptr,
   uint16_t      flags) DART_NOTHROW;
 
+/**
+ * DART allocator used for non-collective global memory allocations using
+ * \ref dart_allocator_alloc similar to \ref dart_memalloc.
+ * An instance of the allocator has to be created using \c dart_allocator_new
+ * before allocating memory using \ref dart_allocator_alloc, which can then
+ * be shared with other units in the same team. The memory has to be free'd
+ * using \ref dart_allocator_free by the unit that allocated it.
+ * The allocator should be destroyed after all allocations have been free'd
+ * using \ref dart_allocator_destroy.
+ */
+typedef struct dart_allocator_struct * dart_allocator_t;
+
+/**
+ * Create a new allocator for non-collective global memory allocations.
+ * This operation is collective among the units in \c team.
+ *
+ * \param pool_size The size of the local memory pool from which global memory
```

Comment:
I guess the pool size is in bytes? That should be mentioned in the documentation. What sizes are allowed? I guess the body allocator requires a 2^n number?